gtsam/navigation/PreintegrationBase.cpp

```diff
@@ -117,23 +117,43 @@ NavState PreintegrationBase::predict(const NavState& state_i,
     OptionalJacobian<9, 6> H2) const {
   Matrix96 D_biasCorrected_bias;
   Vector9 biasCorrected = biasCorrectedDelta(bias_i,
-                                             H2 ? &D_biasCorrected_bias : nullptr);
+      H2 ? &D_biasCorrected_bias : nullptr);
+
+  // Correct for Earth rate
+  NavState state_ii = state_i;
+  if (p().omegaCoriolis){
+    const Vector3 omega = *(p().omegaCoriolis);
+    const Rot3 Gamma_R = Rot3::Expmap(-omega*deltaTij_);
+
+    state_ii = NavState(Gamma_R * state_i.attitude(),
+                        Gamma_R * state_i.position(),
+                        Gamma_R * state_i.velocity());
+    // Jacobian is identity
+  }
 
   // Correct for initial velocity and gravity
   Matrix9 D_delta_state, D_delta_biasCorrected;
-  Vector9 xi = state_i.correctPIM(biasCorrected, deltaTij_, p().n_gravity,
-                                  p().omegaCoriolis, p().use2ndOrderCoriolis, H1 ? &D_delta_state : nullptr,
-                                  H2 ? &D_delta_biasCorrected : nullptr);
+  Vector9 xi = state_ii.correctPIM(biasCorrected, deltaTij_, p().n_gravity,
+      p().omegaCoriolis, p().use2ndOrderCoriolis, H1 ? &D_delta_state : nullptr,
+      H2 ? &D_delta_biasCorrected : nullptr);
 
   // Use retract to get back to NavState manifold
   Matrix9 D_predict_state, D_predict_delta;
-  NavState state_j = state_i.retract(xi,
-                                     H1 ? &D_predict_state : nullptr,
-                                     H1 || H2 ? &D_predict_delta : nullptr);
+  NavState state_j = state_ii.retract(xi, D_predict_state, D_predict_delta);
```

Comment:
We. An select here to use expmap based on integration scheme, once NavState has Lie